base58/src/lib.rs

```diff
@@ -1,19 +1,40 @@
 // SPDX-License-Identifier: CC0-1.0
 
-//! Base58 encoder and decoder.
+//! Bitcoin base58 encoding and decoding.
 //!
-//! This module provides functions for encoding and decoding base58 slices and
-//! strings respectively.
-//!
-
-use core::{fmt, iter, slice, str};
+//! This crate can be used in a no-std environment but requires an allocator.
+
+#![no_std]
+// Experimental features we need.
+#![cfg_attr(docsrs, feature(doc_auto_cfg))]
+#![cfg_attr(bench, feature(test))]
+// Coding conventions.
+#![warn(missing_docs)]
+// Instead of littering the codebase for non-fuzzing code just globally allow.
+#![cfg_attr(fuzzing, allow(dead_code, unused_imports))]
+// Exclude lints we don't think are valuable.
+#![allow(clippy::needless_question_mark)] // https://github.com/rust-bitcoin/rust-bitcoin/pull/2134
+#![allow(clippy::manual_range_contains)] // More readable than clippy's format.
+
+#[macro_use]
+extern crate alloc;
 
-use hashes::{sha256d, Hash};
+#[cfg(feature = "std")]
+extern crate std;
 
-use crate::prelude::*;
+/// Rust implementation of cryptographic hash function algorithms.
+pub extern crate hashes;
```

Comment:
This shouldn't be `pub` and doesn't need a doc.

Reply:
Removed the `extern crate` statement altogether.